changelog/remove-jcb-promotion-leftovers

```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Remove unnecessary residual code related to payment method promotions.
```

Comment:
It's strangely showing me a red minus icon below the changelog entry:

<img width="643" alt="Screenshot 2023-11-01 at 3 59 19 PM" src="https://github.com/Automattic/woocommerce-payments/assets/1288616/786e44a9-3dfa-4f72-adcc-57c16d875244">

Reply:
The issue was due to the absence of an empty newline. Resolved.